### ISSUE TITLE
NVDA 2023.3 beta is extremely sluggish on the Task Manager details tab making it extremely inefficient to navigate

### ISSUE THREAD

**Issue by btman16** [nvaccess/nvda#15503]
### Steps to reproduce:
1. On Windows 10 or 11, open Task Manager and focus the details tab.
2. Arrow through the processes rapidly.
### Actual behavior:

NVDA is extremely slow to respond in this list (it's probably due to the fix for UIA and SysListView32).
### Expected behavior:

NVDA should keep up as I arrow, responding quickly and reading each process, as it does in 2023.2.
#### NVDA installed/portable/running from source:
Installed
#### NVDA version:
2023.3 (all betas)
#### Windows version:
Either Windows 10 or 11, it doesn't matter.
#### Name and version of other software in use when reproducing the issue:
None
#### Does the issue still occur after restarting your computer?
Yes
#### Have you tried any other versions of NVDA? If so, please report their behaviors.
2023.2 does not exhibit the issue.
#### If NVDA add-ons are disabled, is your problem still occurring?
Yes.
#### Does the issue still occur after you run the COM Registration Fixing Tool in NVDA's tools menu?
Yes
[nvda-old.log](https://github.com/nvaccess/nvda/files/12707011/nvda-old.log)

**Comment by amirsol81** [comment]
@btman16 I'm running NVDA Alpha 29323 on Windows 11 22H2 with IBMTTS. The machine has a Core-I7 HQ 10XXX CPU, 16GB of RAM, and an SSD. NVDA is quite responsive in the Details tab of the Task Manager - at least it's as responsive as JAWS, and I get no noticeable lag whatsoever.

**Comment by btman16** [comment]
Hi,

The interesting thing is that some UIA areas are more responsive
(Windows Settings, for example), but then others (typically older
ones) lag a lot.

Thanks,

Brandon

On 9/23/23, amirsol81 ***@***.***> wrote:
> @btman16 I'm running NVDA Alpha 29323 on Windows 11 22H2 with IBMTTS. The
> machine has a Core-I7 HQ 10XXX CPU, 16GB of RAM, and an SSD. NVDA is quite
> responsive in the Details tab of the Task Manager - at least it's as
> responsive as JAWS, and I get no noticeable lag whatsoever.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15503#issuecomment-1732384057
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
“Be what you are. This is the first step towards becoming better than you are.”
– J. C. Hare & A. W. Hare

**Comment by Adriani90** [comment]
Can you check in advanced settings of nvda if registering for UIA events is set on „selective“? Or try to rset that setting to selective. Does that fix your issue?

**Comment by ultrasound1372** [comment]
It's also worth considering what the update speed setting under the view menu is set to. If it's anything other than paused it would generate a flood of repaint events which may very well be the reason NVDA lags in the first place, regardless of registration settings. After all, if you're focused on the list, you would be interested in events pertaining to that list and its children.

**Comment by btman16** [comment]
Hi,

The UIA Registration setting being set to selective makes no
difference, and this issue even happens when the update speed is set
to paused.

Thanks,

Brandon

On 9/23/23, Colton Hill ***@***.***> wrote:
> It's also worth considering what the update speed setting under the view
> menu is set to. If it's anything other than paused it would generate a flood
> of repaint events which may very well be the reason NVDA lags in the first
> place, regardless of registration settings. After all, if you're focused on
> the list, you would be interested in events pertaining to that list and its
> children.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15503#issuecomment-1732418775
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
“Be what you are. This is the first step towards becoming better than you are.”
– J. C. Hare & A. W. Hare

**Comment by Brian1Gaff** [comment]
It must be due to Windows way of displaying it, as on Windows 7, its as snappy as it always was, in case this helps.
 Brian

-- 
***@***.*** 
Sent via blueyonder.(Virgin media) 
Please address personal E-mail to:-
***@***.***, putting 'Brian Gaff'
in the display name field.

**Comment by btman16** [comment]
Hi,

Windows 7's Task Manager doesn't use UIA I don't think, which is why
you probably won't see the issue there.

Thanks,

Brandon

On 9/24/23, Brian Gaff ***@***.***> wrote:
> It must be due to Windows way of displaying it, as on Windows 7, its as
> snappy as it always was, in case this helps.
>  Brian
>
> --
> ***@***.***
> Sent via blueyonder.(Virgin media)
> Please address personal E-mail to:-
> ***@***.***, putting 'Brian Gaff'
> in the display name field.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15503#issuecomment-1732530178
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
“Be what you are. This is the first step towards becoming better than you are.”
– J. C. Hare & A. W. Hare

**Comment by seanbudd** [comment]
I believe this is a duplice of #6245

**Comment by lukaszgo1** [comment]
@btman16  Since you said that this is a regression, to help make sure that this is indeed caused by the change for syslistview32 from MSAA to UIA, could you execute the following in the Python console (NVDA+CTRL+Z), by copying the code below line by line, pasting to the edit field in the Python console, and pressing enter after each command?
```
import UIAHandler
UIAHandler.badUIAWindowClassNames = UIAHandler.badUIAWindowClassNames + ("SysListView32",)
```

Then retry navigating in the task manager.

**Comment by btman16** [comment]
Hi,

Yes, after running that set of commands in the Python Console, it's
super responsive again just like before.

Thanks,

Brandon

On 9/25/23, Łukasz Golonka ***@***.***> wrote:
> @btman16  Since you said that this is a regression, to help make sure that
> this is indeed caused by the change for syslistview32 from MSAA to UIA,
> could you execute the following in the Python console (NVDA+CTRL+Z), by
> copying the code below line by line, pasting to the edit field in the Python
> console, and pressing enter after each command?
> ```
> import UIAHandler
> UIAHandler.badUIAWindowClassNames = UIAHandler.badUIAWindowClassNames +
> ("SysListView32",)
> ```
>
> Then retry navigating in the task manager.
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15503#issuecomment-1733378200
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
“Be what you are. This is the first step towards becoming better than you are.”
– J. C. Hare & A. W. Hare

**Comment by btman16** [comment]
Hi,

It was not responsive on the details tab before I ran that command,
but it was after.

I'm specifically talking about only the details tab here.

Thanks,

Brandon

On 9/25/23, Beqa Gozalishvili ***@***.***> wrote:
> Taskmanager definitely is not supper responsive at all.
>
> Issue #6245 is still exist and nothing was done to improve situation.
>
> On 9/25/23, btman16 ***@***.***> wrote:
>> Hi,
>>
>> Yes, after running that set of commands in the Python Console, it's
>> super responsive again just like before.
>>
>> Thanks,
>>
>> Brandon
>>
>> On 9/25/23, Łukasz Golonka ***@***.***> wrote:
>>> @btman16  Since you said that this is a regression, to help make sure
>>> that
>>> this is indeed caused by the change for syslistview32 from MSAA to UIA,
>>> could you execute the following in the Python console (NVDA+CTRL+Z), by
>>> copying the code below line by line, pasting to the edit field in the
>>> Python
>>> console, and pressing enter after each command?
>>> ```
>>> import UIAHandler
>>> UIAHandler.badUIAWindowClassNames = UIAHandler.badUIAWindowClassNames +
>>> ("SysListView32",)
>>> ```
>>>
>>> Then retry navigating in the task manager.
>>>
>>> --
>>> Reply to this email directly or view it on GitHub:
>>> https://github.com/nvaccess/nvda/issues/15503#issuecomment-1733378200
>>> You are receiving this because you were mentioned.
>>>
>>> Message ID: ***@***.***>
>>
>>
>> --
>> “Be what you are. This is the first step towards becoming better than you
>> are.”
>> – J. C. Hare & A. W. Hare
>>
>>
>> --
>> Reply to this email directly or view it on GitHub:
>> https://github.com/nvaccess/nvda/issues/15503#issuecomment-1733674314
>> You are receiving this because you are subscribed to this thread.
>>
>> Message ID: ***@***.***>
>
>
> --
> with best regards Beqa Gozalishvili
> Tell: +995593454005
> Email: ***@***.***
> Web: https://gozaltech.org
> Skype: beqabeqa473
> Telegram: https://t.me/gozaltech
> facebook: https://facebook.com/gozaltech
> twitter: https://twitter.com/beqabeqa473
> Instagram: https://instagram.com/beqa.gozalishvili
>
>
> --
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/15503#issuecomment-1733795505
> You are receiving this because you were mentioned.
>
> Message ID: ***@***.***>


-- 
“Be what you are. This is the first step towards becoming better than you are.”
– J. C. Hare & A. W. Hare

**Comment by TheQuinbox** [comment]
I'm also fully able to reproduce this with latest alpha, on a 16-core machine, so it's definitely not a processor issue. And, same as @btman16, that Python Console hack fixes it temporarily.

**Comment by seanbudd** [comment]
Please confirm the alpha version you are using as this regression should definitely be fixed, only #6245 should be relevant at this time

**Comment by seanbudd** [comment]
@TheQuinbox - do you also have an issue with Task Manager with NVDA 2023.2? if so, #6245 is the related issue not this one
